src/Core/src/Platform/Windows/MauiWinUIWindow.cs

```diff
@@ -87,6 +87,9 @@ private void OnClosedPrivate(object sender, UI.Xaml.WindowEventArgs args)
 		{
 			OnClosed(sender, args);
 
+			Activated -= OnActivated;
```

Comment:
After a bunch of trial and error, this was the solution to getting the weak reference to be marked not-alive.